### PR TITLE
Build: Auto-import correction

### PR DESCRIPTION
**Pull request by ndelangen**
## What I did

In VScode the auto-import feature often imported paths like these:
`code/lib/types` or `code/core/types`.
.. these paths do not work.

I've adjusted the TS config, so TypeScript should be able to find & suggest the correct paths.

It kinda refuses to auto-import from `storybook/internal/X` and would rather import from `@storybook/core/X` instead.
This however is not the path we want either.

But that path can be somewhat easily converted by hand..
But you won't have to, because I wrote an ESlint rule that automatically does it for you!

It also found an issue here:
https://github.com/storybookjs/storybook/blob/d20eab73d02709f631162eeeedf7cc4af80c5d12/code/lib/blocks/src/blocks/Source.tsx#L3-L4

I simplified the `tsconfig`.json` files a bit. (`strict`-mode was already the default, in the base config, but was still setup in most packages for example)

I had to change the vue3 test-storybook from `portal:` to `file:` protocol, because something for that specific project made it so the `paths` property in it's `package.json` was conflicting. It yielded this error in CI:
https://app.circleci.com/pipelines/github/storybookjs/storybook/79345/workflows/f27624ed-dc55-4a51-82ce-bcc76fb24b83/jobs/676050

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

## Greptile Summary

- Updated TypeScript configurations to correct auto-import paths (`tsconfig.json`).
- Added `beforeAll` hooks and corresponding tests (`code/core/src/preview-api/modules/store/csf/beforeAll.ts`, `beforeAll.test.ts`).
- Removed `node-fetch` imports and dependencies across multiple files (`scripts/package.json`, `scripts/release/*`, `scripts/utils/*`).
- Modified key property in `MethodCall.tsx` for better React rendering performance.
- Updated README.md to display individual sponsor links.

### CODE REVIEW

File: scripts/task.ts [hunk 1]
```diff
@@ -473,8 +473,8 @@ async function run() {
         if (controller) controllers.push(controller);
       } catch (err) {
         invariant(err instanceof Error);
-        logger.error(`Error running task ${getTaskKey(task)}:`);
-        logger.error(JSON.stringify(err, null, 2));
+        // logger.error(`Error running task ${getTaskKey(task)}:`);
```
Comment (valentinpalkovic) [inline scripts/task.ts:476]: Was this commented out for testing purposes?

File: test-storybooks/portable-stories-kitchen-sink/vue3/package.json [hunk 2]
```diff
@@ -16,66 +16,66 @@
...
-    "@storybook/svelte-webpack5": "portal:../../../code/frameworks/svelte-webpack5",
-    "@storybook/sveltekit": "portal:../../../code/frameworks/sveltekit",
-    "@storybook/test": "portal:../../../code/lib/test",
-    "@storybook/vue3": "portal:../../../code/renderers/vue3",
-    "@storybook/vue3-vite": "portal:../../../code/frameworks/vue3-vite",
-    "@storybook/vue3-webpack5": "portal:../../../code/frameworks/vue3-webpack5",
-    "@storybook/web-components": "portal:../../../code/renderers/web-components",
-    "@storybook/web-components-vite": "portal:../../../code/frameworks/web-components-vite",
-    "@storybook/web-components-webpack5": "portal:../../../code/frameworks/web-components-webpack5",
+    "@storybook/addon-a11y": "file:../../../code/addons/a11y",
```
Comment (valentinpalkovic) [inline test-storybooks/portable-stories-kitchen-sink/vue3/package.json:19]: should we change all portable stories kitchen sinks to use `link` instead of `portal`?

File: code/addons/links/tsconfig.json [hunk 3]
```diff
@@ -1,8 +1,11 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
-    "strict": true,
-    "skipLibCheck": true
+    "baseUrl": ".",
+    "paths": {
+      "storybook/internal/*": ["../../lib/cli/core/*"],
+      "react": ["../../node_modules/react"]
```
Comment (valentinpalkovic) [inline code/addons/links/tsconfig.json:7]: Why is this necessary?

**Review by valentinpalkovic — APPROVED** [review]

File: code/addons/links/tsconfig.json (hunk 3 above)
Comment (ndelangen) [reply code/addons/links/tsconfig.json:7]: Yeah.. please help me out removing it somehow...

TypeScript got things wrong, badly. and though this import referred to the `@storybook/addon-links/react` export.
Which is obviously very wrong, and TypeScript complained that it doesn't export any of the types needed.

I made TypeScript realize the correct path like this, but I'd love to remove this as well.

File: test-storybooks/portable-stories-kitchen-sink/vue3/package.json (hunk 2 above)
Comment (ndelangen) [reply test-storybooks/portable-stories-kitchen-sink/vue3/package.json:19]: Well, it's very inconvenient for whoever needs to work on these if they are `file:` protocol, so I figured if they aren't broke, don't fix em.

@yannbf is the person working on these most, so I'll defer the decision if we choose consistency over ease of maintenance, to him.

I'm ok with either option.

Comment (yannbf) [reply test-storybooks/portable-stories-kitchen-sink/vue3/package.json:19]: I'm OK with whatever change

**Review by yannbf — APPROVED** [review]

**Review by kasperpeulen — APPROVED** [review]

File: code/tsconfig.json [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
+    "incremental": false,
+    "isolatedModules": true,
     "jsx": "react",
-    "moduleResolution": "Node",
-    "target": "ES2020",
+    "lib": ["dom", "dom.iterable", "esnext"],
     "module": "CommonJS",
+    "moduleResolution": "Node",
+    "noImplicitAny": true,
+    "noUnusedLocals": false,
```
Comment (kasperpeulen) [inline code/tsconfig.json:16]: Why did we change the `noUnusedLocals` setting?

Comment (ndelangen) [reply code/tsconfig.json:16]: I changed it, because it's making CI fail over an unused variable.

Something that gets bundled out anyway.
It's just not something worth failing a whole build/compile over.

I'm interested in adding an eslint rule for it though.

Comment (kasperpeulen) [reply code/tsconfig.json:16]: fair enough
